thavalon-server/src/database/mod.rs

```diff
@@ -0,0 +1,16 @@
+use mongodb::{options::ClientOptions, Client, Database};
+
+mod accounts;
+pub use self::accounts::*;
+
+const MONGO_HOST: &str = "mongodb://admin:secret@database:27017";
+const THAVALON_DB: &str = "thavalon_db";
+
+/// Gets a MongoDB client connection.
+/// Since the database is on the same cluster (?), failing here should cause a crash.
+async fn get_db_client() -> Database {
```

Comment:
Could we hold on to the DB connection instead of reconnecting on every request? I think the official mongo driver supports connection pooling

Reply:
So...this one has been a challenge. Take a look at the new mod.rs for database. I'm not wild about what I came up with, but it's what I thought of given the constraints we have:

- Can't use lazy_static to initialize a static connection, because mongo requires async and tokio isn't running then.
- Static muts are unsafe
- Locks can't be held across an await, even when they aren't used by that await or are dropped. This leads to some weird scoping things that come up.

I'm open to suggestions on this one, as my solution isn't very good, I don't think.

Reply:
I've also coded it using two unsafes{} and it's much cleaner...I may go with that option. Essentially we have one fn to initialize the connection that uses unsafe{} to save to a static mut. Then we have another function that uses unsafe{} to access it.

Reply:
Jared has come up with a better solution which I've switched to. Everything seems to work, but now I'm getting this warning which I've never seen, and Google has failed me.

[os_info::imp::lsb_release] lsb_release command failed with Os { code: 2, kind: NotFound, message: "No such file or directory" }

Everything still seems to work, so I pushed it.

Reply:
That... is an odd error. lsb_release is a command that prints out info about the running Linux distribution, but not all distributions have it. No idea what would be running it though.